flake.nix

```diff
@@ -38,63 +36,75 @@
         };
       };
 
-      flake.lib = import ./lib { inherit lib; };
+      lib = import ./lib { inherit lib; };
 
-      perSystem =
-        {
-          config,
-          pkgs,
-          self',
-          ...
-        }:
+      templates = {
+        flake-parts = {
+          description = "Example flake with nix-unit and flake-parts";
+          path = ./templates/flake-parts;
+        };
+      };
+
+      # TODO: Figure out how to distribute flake-parts without using flake-parts on the top-level
+      modules = throw "Dunno lol";
+
+      packages = forAllSystems (
+        system:
         let
-          inherit (pkgs) stdenv;
+          pkgs = nixpkgs.legacyPackages.${system};
           drvArgs = {
             nix = pkgs.nixVersions.nix_2_24;
           };
         in
         {
-          packages.nix-unit = pkgs.callPackage ./default.nix drvArgs;
-          packages.default = self'.packages.nix-unit;
-          packages.doc = pkgs.callPackage ./doc {
+          nix-unit = pkgs.callPackage ./default.nix drvArgs;
+          default = self.packages.${system}.nix-unit;
+          doc = pkgs.callPackage ./doc {
             inherit self;
           };
-          devShells.default =
+        }
+      );
+
+      formatter = forAllSystems (
+        system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
+        in
+        (treefmt-nix.lib.evalModule pkgs ./dev/treefmt.nix).config.build.wrapper
+      );
+
+      devShells = forAllSystems (
+        system:
+        let
+          pkgs = nixpkgs.legacyPackages.${system};
+          inherit (pkgs) stdenv;
+          drvArgs = {
+            nix = pkgs.nixVersions.nix_2_24;
+          };
+        in
+        {
+          default =
             let
               pythonEnv = pkgs.python3.withPackages (_ps: [ ]);
             in
             pkgs.mkShell {
-              nativeBuildInputs = self'.packages.nix-unit.nativeBuildInputs ++ [
+              nativeBuildInputs = self.packages.${system}.nix-unit.nativeBuildInputs ++ [
                 pythonEnv
                 pkgs.difftastic
                 pkgs.nixdoc
                 pkgs.mdbook
                 pkgs.mdbook-open-on-gh
                 pkgs.mdbook-cmdrun
-                config.treefmt.build.wrapper
+                self.formatter.${system}
               ];
-              inherit (self'.packages.nix-unit) buildInputs;
+              inherit (self.packages.${system}.nix-unit) buildInputs;
               shellHook = lib.optionalString stdenv.isLinux ''
                 export NIX_DEBUG_INFO_DIRS="${pkgs.curl.debug}/lib/debug:${drvArgs.nix.debug}/lib/debug''${NIX_DEBUG_INFO_DIRS:+:$NIX_DEBUG_INFO_DIRS}"
                 export NIX_UNIT_OUTPATH=${self}
               '';
             };
-        };
+        }
+      );
 
-      # Extra things to load only when accessing development-specific attributes
-      # such as `checks`
-      partitionedAttrs.checks = "dev";
```

Comment:
Removing this line disconnects the check derivation for the flake-parts module.

Considering that subflakes still are still quite broken until the next NixOS release, it still has to be wired through `/flake.nix`.
You'll need `flake-compat` to load the inputs from `dev` and then cherry-pick the relevant attributes from a `mkFlake` call that resembles a combination of `module` below.

(I guess that's a partial answer to your question about the added value here. `flake-parts.flakeModules.partitions` is a ready to use solution for this.)

Reply:
> subflakes still are still quite broken until the next NixOS release

where can I find out about the upcoming improvements? Thanks!